### DISCUSSION TITLE
HTTP request timeout for EdgeX Demonstration API Walk Through (Minnesota)

### DISCUSSION THREAD

**Discussion by dbruno74** [edgexfoundry/community#125] [Q&A]
I'm following the tutorial [here](https://docs.edgexfoundry.org/3.0/walk-through/Ch-Walkthrough/) to define, register and provision a device in edgex Minnesota.
When [provisioning the device](https://docs.edgexfoundry.org/3.0/walk-through/Ch-WalkthroughProvision/), the curl call fails with
`HTTP request timeout`.

I tried the tutorial with EdgeX 2.3 and it worked like a charm.

I compared the code for [v2.3.0](https://github.com/edgexfoundry/edgex-go/blob/v2.3.0/internal/core/metadata/application/notify.go#L64) with [v3.0.0](https://github.com/edgexfoundry/edgex-go/blob/v3.0.0/internal/core/metadata/application/notify.go#L26) and I found that in v3.0.0, in case of failure of Device Validation request, an error is issued instead of a warning, letting the whole function fails.

Could you help me to understand the reason of such failure and eventually how to workaround it?

**Comment by farshidtz** [comment]
It seems like Core Matadata is waiting for the validation response over the Message Bus, but in this case, there is no device service that can respond:
`2023-06-09T11:42:38+02:00 edgexfoundry.core-metadata[26101]: level=ERROR ts=2023-06-09T09:42:38.229407254Z app=core-metadata source=device.go:67 X-Correlation-ID=5ad239e0-640e-465c-9b1c-feb3e176fd84 msg="Error sending request to topic 'edgex/camera-control-device-service/validate/device' -> timed out waiting for response on edgex/response/camera-control-device-service/79352a47-82a8-41e0-a902-6a273d7567af topic"`
`2023-06-09T11:42:38+02:00 edgexfoundry.core-metadata[26101]: level=ERROR ts=2023-06-09T09:42:38.22955417Z app=core-metadata source=encoding.go:33 msg="Error encoding the data: http: Handler timeout"`

Based on the statement below, I expected that setting `adminState` to `LOCKED` would have disabled the validation request, but that wasn't the case.
> When a device service is set to LOCKED, it is not suppose to respond to any command requests nor send data from the devices. [[ref](https://docs.edgexfoundry.org/3.0/walk-through/Ch-WalkthroughDeviceService/)]

Maybe there is a way to explicitly disable validation.

**Comment by lenny-goodell** [comment]
This walk-thru is now broken with the new use of message bus to request validation since the walk thru doesn't use an actual device service that is running.
